src/editor.c

```diff
@@ -857,13 +856,12 @@ static void on_char_added(GeanyEditor *editor, SCNotification *nt)
 		case ':':	/* C/C++ class:: syntax */
 		/* tag autocompletion */
 		default:
-#if 0
-			if (! editor_start_auto_complete(editor, pos, FALSE))
-				request_reshowing_calltip(nt);
-#else
 			editor_start_auto_complete(editor, pos, FALSE);
-#endif
 	}
+
+	plugin_extension_calltips_show(editor->document, FALSE);
+	plugin_extension_autocomplete_perform(editor->document, FALSE);
```

Comment:
Hum… do we actually want the calls here?  I'm fine with them, don't get me wrong, but does this actually help the extensions, or do they have to listen to `editor-notify` anyway and this becomes kind of artificial?

Reply:
Technically they could be obtained from `editor-notify` and plugins could call `plugin_extension_autocomplete_provided()` and `plugin_extension_calltips_provided()` to check it is them who executes them. But then you have the calls where the `force` parameter is `TRUE` and these are still needed. So you'd end up with something like
```
plugin_extension_calltips_show_force(GeanyDocument *doc);
plugin_extension_autocomplete_perform_force(GeanyDocument *doc);
```
inside the API and the non-force variants would have to be handled in a separate way outside the API which is quite strange IMO. So I'd leave them as they are.

Reply:
OK, if you think extensions implementing those will basically never have to re do this separately because they have some other logic to when it should happen let's keep it that way. It's pretty reasonable to check this upon character insertion I guess, and easy enough for an extension to ignore the force=FALSE cases if it really doesn't want to do it that way I guess.

Reply:
Well, I have just 2 real-world examples to base this on - the LSP plugin and Geany itself. For Geany, both calltips and and autocomplete originate in `on_char_added()` (except the `force` variants triggered from elsewhere). For calltips, LSP uses "trigger characters" which, when typed, trigger the calltip display (it's typically `(` and `,`). Autocompletion also happens as you type and what we call "scope autocompletion" is also based on trigger characters.

One could think of some kind of autocompletion also happen when e.g. deleting characters but I think it would be annoying and don't think anyone does that.

So I think it is reasonably safe to assume `on_char_added()` is sufficient.